modules/dot-prefix-validation/src/main/java/org/elasticsearch/validation/DotPrefixValidator.java

```diff
@@ -73,7 +73,8 @@ public abstract class DotPrefixValidator<RequestType> implements MappedActionFil
             "\\.ml-state-\\d+",
             "\\.slo-observability\\.sli-v\\d+.*",
             "\\.slo-observability\\.summary-v\\d+.*",
-            "\\.entities\\.v\\d+\\.latest\\..*"
+            "\\.entities\\.v\\d+\\.latest\\..*",
+            "\\.monitoring-es-8-.*"
```

Comment:
I would probably add an exception for `.monitoring-.*`, as `es` is just one of them. Metricbeat can send data to [3 more data streams](https://github.com/elastic/beats/blob/main/metricbeat/helper/elastic/elastic_test.go#L41-L56), namely:
* `.monitoring-logstash-8-mb`
* `.monitoring-kibana-8-mb`
* `.monitoring-beats-8-mb`

And potentially, Metricbeat 8 could send monitoring data from an old Enterprise Search 8 on ES 9 into `.monitoring-ent-search-8-mb`